r/azurerm_windows_function_app - import not working

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->
Terraform v0.12.31
+ provider.azurerm v3.0.1
+ provider.random v3.1.2

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_windows_function_app`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
provider "azurerm" {
  features {}
}

resource "random_pet" "example" {
  length = 1
}

resource "azurerm_resource_group" "example" {
  name     = "azure-functions-test-rg"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                     = format("%stestsa", random_pet.example.id)
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_app_service_plan" "example" {
  name                = "azure-functions-test-service-plan"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  sku {
    tier = "Standard"
    size = "S1"
  }
}

resource "azurerm_function_app" "example" {
  name                       = format("%stest-azure-functions", random_pet.example.id)
  location                   = azurerm_resource_group.example.location
  resource_group_name        = azurerm_resource_group.example.name
  app_service_plan_id        = azurerm_app_service_plan.example.id
  storage_account_name       = azurerm_storage_account.example.name
  storage_account_access_key = azurerm_storage_account.example.primary_access_key
}

output "id" {
  value = azurerm_function_app.example.id
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behaviour

<!--- What should have happened? --->
Import succeeded.
### Actual Behaviour

<!--- What actually happened? --->
```hcl
Error: specified Service Plan is not a Windows Functionapp plan
```
### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply -auto-approve`
2. `terraform state rm azurerm_function_app.example`
3. `rename azurerm_function_app => azurerm_windows_function_app`
4. `terraform import azurerm_windows_function_app.example <id from the output>`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

I set TF_LOG to debug and to troubleshoot this.  Just before the error is received, a GET is made to get info on the Service Plan.  

In my case, this said:

```
{"id":"/subscriptions/xxxxxxxx-54a4-40bd-a55b-95de04807789/resourceGroups/holx-devopssbx-iot-dev/providers/Microsoft.Web/serverfarms/holx-devopssbx-iot-slinkreg-ASP-dev","name":"holx-devopssbx-iot-slinkreg-ASP-dev","type":"Microsoft.Web/serverfarms","kind":"Windows","location":"East US","properties":{"serverFarmId":43937,"name":"holx-devopssbx-iot-slinkreg-ASP-dev","workerSize":"Default","workerSizeId":0,"workerTierName":null,"numberOfWorkers":1,"currentWorkerSize":"Default","currentWorkerSizeId":0,"currentNumberOfWorkers":1,"status":"Ready","webSpace":"holx-devopssbx-iot-dev-EastUSwebspace","subscription":"xxxxxxxx-54a4-40bd-a55b-95de04807789","adminSiteName":null,"hostingEnvironment":null,"hostingEnvironmentProfile":null,"maximumNumberOfWorkers":10,"planName":"VirtualDedicatedPlan","adminRuntimeSiteName":null,"computeMode":"Dedicated","siteMode":null,"geoRegion":"East US","perSiteScaling":false,"elasticScaleEnabled":false,"maximumElasticWorkerCount":1,"numberOfSites":5,"hostingEnvironmentId":null,"isSpot":false,"spotExpirationTime":null,"freeOfferExpirationTime":null,"kind":"Windows","resourceGroup":"holx-devopssbx-iot-dev","reserved":false,"isXenon":false,"hyperV":false,"mdmId":"waws-prod-blu-083_43937","targetWorkerCount":0,"targetWorkerSizeId":0,"provisioningState":"Succeeded","webSiteId":null,"existingServerFarmIds":null,"kubeEnvironmentProfile":null,"zoneRedundant":false},"sku":{"name":"S1","tier":"Standard","size":"S1","family":"S","capacity":1}}: timestamp=2022-03-27T08:00:16.171-0700
2022-03-27T08:00:16.211-0700 [ERROR] vertex "azurerm_windows_function_app.funcapp1 (import id \"/subscriptions/xxxxxxxx-54a4-40bd-a55b-95de04807789/resourceGroups/holx-devopssbx-iot-dev/providers/Microsoft.Web/sites/holx-devopssbx-iot-slinkreg-FunctApp-dev\")" error: specified Service Plan is not a Windows Functionapp plan
[31m[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mspecified Service Plan is not a Windows Functionapp plan[0m

```

In there one of the attributes was `"kind":"Windows"`


I then looked at source for the provider. 

In file https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/appservice/windows_function_app_resource.go

is this:

```
if strings.Contains(strings.ToLower(*sp.Kind), "linux") || !(strings.Contains(strings.ToLower(*sp.Kind), "elastic") || strings.Contains(strings.ToLower(*sp.Kind), "functionapp") || strings.Contains(strings.ToLower(*sp.Kind), "app")) {
			return fmt.Errorf("specified Service Plan is not a Windows Functionapp plan")
		}

```


I'm not a Go programmer, but this looks like a bunch of "or" conditions.   The second expression looks to be saying if "Kind is not elastic".  So as my kind of "Windows" evaluates true here and incorrectly errors out.

--- a/internal/services/appservice/windows_function_app_resource.go
+++ b/internal/services/appservice/windows_function_app_resource.go
@@ -826,7 +826,8 @@ func (r WindowsFunctionAppResource) CustomImporter() sdk.ResourceRunFunc {
 		if err != nil || sp.Kind == nil {
 			return fmt.Errorf("reading Service Plan for Windows %s: %+v", id, err)
 		}
-		if strings.Contains(strings.ToLower(*sp.Kind), "linux") || !(strings.Contains(strings.ToLower(*sp.Kind), "elastic") || strings.Contains(strings.ToLower(*sp.Kind), "functionapp") || strings.Contains(strings.ToLower(*sp.Kind), "app")) {
+
+		if strings.Contains(strings.ToLower(*sp.Kind), "linux") {
 			return fmt.Errorf("specified Service Plan is not a Windows Functionapp plan")
 		}
 

